components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/IdentityKeyStoreResolver.java

```diff
@@ -0,0 +1,612 @@
+import org.wso2.carbon.identity.core.util.IdentityConfigParser;
+import org.wso2.carbon.identity.core.util.IdentityKeyStoreResolverConstants.InboundProtocol;
+import org.wso2.carbon.identity.core.util.IdentityKeyStoreResolverException;
+import org.wso2.carbon.identity.core.util.IdentityKeyStoreResolverUtil;
+import org.wso2.carbon.identity.core.util.IdentityTenantUtil;
+import org.wso2.carbon.utils.CarbonUtils;
+
+import java.io.File;
+import java.security.Key;
+import java.security.KeyStore;
+import java.security.cert.Certificate;
+import java.security.interfaces.RSAPublicKey;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static org.wso2.carbon.identity.core.util.IdentityKeyStoreResolverConstants.ATTR_NAME_KEYSTORE_NAME;
+import static org.wso2.carbon.identity.core.util.IdentityKeyStoreResolverConstants.ATTR_NAME_PROTOCOL;
+import static org.wso2.carbon.identity.core.util.IdentityKeyStoreResolverConstants.ATTR_NAME_USE_IN_ALL_TENANTS;
+import static org.wso2.carbon.identity.core.util.IdentityKeyStoreResolverConstants.CONFIG_ELEM_KEYSTORE_MAPPING;
+import static org.wso2.carbon.identity.core.util.IdentityKeyStoreResolverConstants.CONFIG_ELEM_KEYSTORE_MAPPINGS;
+import static org.wso2.carbon.identity.core.util.IdentityKeyStoreResolverConstants.CONFIG_ELEM_SECURITY;
+import static org.wso2.carbon.identity.core.util.IdentityKeyStoreResolverConstants.ErrorMessages;
+import static org.wso2.carbon.identity.core.util.IdentityKeyStoreResolverConstants.PRIMARY_KEYSTORE_CONFIG_PATH;
+
+
+/**
+ * KeyStore manager for inbound authentication protocols.
+ * Retrieve keystores, private keys, public keys and public certificates.
+ */
+public class IdentityKeyStoreResolver {
+
+    private static IdentityKeyStoreResolver instance = null;
+
+    private static ConcurrentHashMap<InboundProtocol, IdentityKeyStoreMapping>
+            keyStoreMappings = new ConcurrentHashMap<>();
+
+    // Hashmaps to store retrieved private keys and certificates.
+    // This will reduce the time required to read configs and load data from keystores everytime.
+    private static Map<String, Key> privateKeys = new ConcurrentHashMap<>();
+    private static Map<String, Certificate> publicCerts = new ConcurrentHashMap<>();
+
+    private static final Log LOG = LogFactory.getLog(IdentityKeyStoreResolver.class);
+
+    private IdentityKeyStoreResolver() {
+
+        parseIdentityKeyStoreMappingConfigs();
+    }
+
+    public static IdentityKeyStoreResolver getInstance() {
+
+        if (instance == null) {
+            instance = new IdentityKeyStoreResolver();
+        }
+        return instance;
+    }
+
+    /**
+     * Return Primary or tenant keystore according to given tenant domain.
+     *
+     * @param tenantDomain  Tenant domain
+     * @return Primary or tenant keystore
+     * @throws IdentityKeyStoreResolverException the exception in the IdentityKeyStoreResolver class
+     */
```

Comment:
```suggestion
    /**
     * Return Primary or tenant keystore according to given tenant domain.
     *
     * @param tenantDomain  Tenant domain.
     * @return Primary or tenant keystore.
     * @throws IdentityKeyStoreResolverException the exception in the IdentityKeyStoreResolver class.
     */
```

Reply:
Fixed in https://github.com/wso2/carbon-identity-framework/pull/5794/commits/3579016060a0e2c08cc4dfb4b3a3cf1e68d0ff8d